statsmodels/base/model.py

```diff
@@ -443,8 +443,13 @@ def fit(self, start_params=None, method='newton', maxiter=100,
             Hinv = np.linalg.inv(-retvals['Hessian']) / nobs
         elif not skip_hessian:
             try:
-                Hinv = np.linalg.inv(-1 * self.hessian(xopt))
-            except:
+                H = -1 * self.hessian(xopt)
+                w, v = np.linalg.eigh(H)
+                if np.min(w) <= 0:
+                    raise RuntimeError
+                Hinv = v.dot(np.diag(1.0 / w)).dot(v.T)
+                Hinv = np.asfortranarray((Hinv + Hinv.T) / 2.0)
+            except RuntimeError:
```

Comment:
there is no reason anymore to use a try ... except
The warning and alternative can just go below the `if np.min(w) <= 0` and else